src/book/02-system/04-hardware/04-dynamixel-protocol.mdx

```diff
@@ -0,0 +1,388 @@
+
+</Alert>
+
+A UART has two pins, an input called RX, and an output called TX.
+
+![Two abstract UARTs](./images/dynamixel_protocol/pure_uart.drawio.png 'Two abstract UARTs have their RX-pins and TX-pins connected each way.')
+
+The term UART can also refer to the way how data is framed. If one will, it can almost be informally thought of as a very low-level and very basic protocol. Here, UART can refer to the special bits that are needed to frame data, all of which need to be agreed upon between both devices on either end.
+
+![An 8N1 frame has one start-bit as always, eight bits of data, no parity-bit, and one stop-bit.](./images/dynamixel_protocol/uart_frame.drawio.png 'An 8N1 frame has one start-bit as always, eight bits of data, no parity-bit, and one stop-bit.')
+
+| Line        | Logic-level | Description                                                                 | Optional |
+| ----------- | ----------- | --------------------------------------------------------------------------- | -------- |
+| Idle        | 1           | is where there is no data.                                                  | no       |
+| Start-bit   | 0           | is the beginning of the data.                                               | no       |
+| Data-bits   | ?           | are the bits of data from the least significant to the most significant.    | no       |
+| Parity-bits | ?           | tell whether there has been any corruption of the data during transmission. | yes      |
+| Stop-bit    | 1           | is the end of the data.                                                     | yes      |
+
+The receiver looks for the start-bit's falling edge. When it has found the edge, it begins to time and to sample each bit thereforth based on the agreed upon bitrate. Most setups are written by three symbols, e.g. 8N1 means that there are eight data-bits, no parity bits, and one stop-bit. Both Dynamixel protocols and many other serial ones are 8N1. Henceforth in this guide, this is the assumed framing.
+
+<Alert type="warn">
+
+Two UARTs must agree on the baudrate, framing (e.g. 8N1), and the signal-levels in order to communicate.
+
+</Alert>
+
+However, pure UART often only refers to abstract bits or logic levels, not voltages. Signalling techniques such as RS485, RS232, and TTL are what are further needed for a complete system to define the electrical behaviour as well as maybe flow-control (e.g. RS232).
+
+## Duplexity
+
+Firstly, one must understand the concept of duplexity since the half duplexity of the Dynamixel protocol is not always intuitive for beginners. There are three kinds of digital communication, namely simplex, full duplex, and half duplex.
```

Comment:
```suggestion
There are three kinds of digital communication: simplex, full duplex, and half duplex. Duplexity is important to understand because the Dynamixel protocol uses half duplexity, which is not always intuitive for beginners.
```